src/main.rs

```diff
@@ -164,3 +152,54 @@ fn pretty_error(err: &failure::Error) -> String {
     }
     pretty
 }
+
+fn get_watchman_query_v1(git_work_tree: &std::path::Path, token: String) -> Value {
+    let epoch_seconds: u64 = token.parse().unwrap_or(0) / 1_000_000_000;
+    json!(
+         [
+             "query",
+             git_work_tree,
+             {
+                 "since": epoch_seconds,
+                 "fields": ["name"],
+                 "expression": [
+                     "not", [
+                         "allof",[
+                             "since",
+                             epoch_seconds,
+                             "cclock"
+                         ],
+                         [
+                             "not",
+                             "exists"
+                         ]
+                     ]
+                 ]
+             }
+        ]
+    )
+}
+
+fn get_watchman_query_v2(git_work_tree: &std::path::Path, token: String) -> Value {
+    // the token following `since` expression can be either epoch second as integer or a clock id as string
```

Comment:
Have you gotten the clock id as a response from git fsmonitor yet? I keep getting the epoch timestamp. I've set it to version 2, and am using `watchman clock .` as the response to the monitor. Is there something I'm missing?

Reply:
Yes. need to make changes to git index so it'll get updated, like `git add xxx`. To reset, can delete `.git/index`